src/backend/joanie/core/api/client.py

```diff
@@ -186,6 +186,29 @@ def _get_organization_with_least_active_orders(self, product, course):
 
         return organizations
 
+    def _is_order_count_above_max(self, course, product):
+        """Check if number of validated orders is above max_validated_orders"""
+        validated_order_count = (
+            models.Order.objects.filter(
+                product=product,
+                course=course,
+            )
+            .filter(
+                Q(state=enums.ORDER_STATE_VALIDATED)
+                | Q(state=enums.ORDER_STATE_PENDING)
+            )
+            .count()
+        )
+        try:
+            max_validated_orders = product.course_relations.get(
+                course=course
+            ).max_validated_orders
+        except models.CourseProductRelation.DoesNotExist:
+            max_validated_orders = 0
+        return (
+            max_validated_orders != 0 and validated_order_count >= max_validated_orders
+        )
```

Comment:
You could use annotation to optimize db access.
```suggestion
        product_course_relation_query = models.CourseProductRelation.objects.filter(
            course=OuterRef("course"), product=OuterRef("product")
        ).values("max_validated_orders")[:1]

        annotation = (
            models.Order.objects.filter(
                product=product, course=course,
                state__in=(enums.ORDER_STATE_VALIDATED, enums.ORDER_STATE_PENDING)
            )
            .annotate(
                max_validated_orders=Subquery(product_course_relation_query),
                orders_count=Count('id', distinct=True)
            )
            .values("max_validated_orders", "orders_count").first() or {}
        )

        max_validated_orders = annotation.get("max_validated_orders", 0)
        validated_order_count = annotation.get("orders_count", 0)
```